### PR TITLE
Fixes #435 :- Full screen mode

### LINKED ISSUE #429: Adding customization menu for media walls.
Currently, we have a basic prototype ready. Next thing to do is to make customization menu to help user personalize the media wall. In the first step, we would adding option to change just the search term. 

screen shot 2017-06-27 at 6 45 30 am

Please suggest @hemantjadon @mariobehling @sudheesh001 .

### LINKED ISSUE #435: Adding Full screen Option for media walls
Idea is to add full-screen option for media walls.

### PR DESCRIPTION
**Pull request by Achint08**
**Changes proposed in this pull request**
- Adds Fullscreen option
- Adds dialog box for customization menu implemetation.

**Screenshots (if appropriate)** 

screen shot 2017-07-04 at 7 46 41 am

**Link to live demo (if appropriate):** 
https://achint08.github.io/loklak_search/

**Closes #435  **

**Note**

Gh-pages show the basic prototype of customization menu, however, the option will be disabled in the deployment until some stable version of setting is made.

### CODE REVIEW

File: src/app/media-wall/media-wall-header/media-wall-header.component.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
 
 @Component({
 	selector: 'media-wall-header',
 	templateUrl: './media-wall-header.component.html',
 	styleUrls: ['./media-wall-header.component.scss']
 })
 export class MediaWallHeaderComponent implements OnInit {
-		@Input() query: string;
+	@Input() query: string;
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-header/media-wall-header.component.ts]: Indentation error

File: src/app/media-wall/media-wall.module.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { LoklakMediaWallRoutingModule } from './media-wall-routing.module';
 import { MediaWallCardComponent } from './media-wall-card/media-wall-card.component';
 import { MediaWallLinkerComponent } from './media-wall-linker/media-wall-linker.component';
 import { MediaWallHeaderComponent } from './media-wall-header/media-wall-header.component';
+import { ToggleFullscreenDirective } from '../shared//full-screen.directive';
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall.module.ts]: The path is not correct path why double `//` add this to `index.ts` file of shared folder and use directly

**Comment by Achint08** [comment]
Please review @hemantjadon @vibhcool @singhpratyush @djmgit @kavithaenair @SKrPl

**Comment by hemantjadon** [comment]
@Achint08 this looks good to me Just few points the tabs look great I guess 
Use the theme color for the tab accent.

**Comment by Achint08** [comment]
@hemantjadon Done. Theming for tabs is not available. Done this using `\deep\` currently. Will adjust colouring when we will proceed with theming components.

**Comment by mariobehling** [comment]
Instead of providing a big settings symbol, please provide a less intrusive way, e.g. a very small icon that show the settings symbol on scroll over.

**Comment by mariobehling** [comment]
As the customization menu does not provide any functionality yet, I don't understand why it should be merged. The only functionality added I see is full screen. Please provide a separate full screen icon, that only shows up on scroll over and get this in first. Regarding customization: Only add areas where there is a feature available. Please avoid "under construction".

**Comment by Achint08** [comment]
@mariobehling To avoid long PRs, I am just adding a UI. And as I mentioned in the Description before, option for customization will be disabled till the next PR which will add settings for sure.

**Comment by mariobehling** [comment]
> To avoid long PRs, I am just adding the UI.

We are not in 1998 when we show "Under Construction" sites. If you want to make the popup, then implement one function and start with this one.

**Comment by Achint08** [comment]
@mariobehling 
Please Checkout now. https://achint08.github.io/loklak_search/wall?query=fossasia
I have currently added an input to the customization menu for changing the query. Will be adding more options like profanity filter, advanced search and other customizations in the following PRs.

**Comment by mariobehling** [comment]
Search "customization" is actually just search, but you could add additional facets here. Before this features goes into the media wall it should go into an advanced search page like https://twitter.com/search-advanced. Please follow up in this issue here: https://github.com/fossasia/loklak_search/issues/447

The implementation is buggy, e.g. scroll over config button on page shows tool tips off-screen. In general the buttons should be hidden on the live site anyways and only appear on scroll over.

**Comment by Achint08** [comment]
@mariobehling @hemantjadon  Please review now. :)
Actually, what I felt is that keeping division hidden on a live site and showing up only on scroll-over might keep users unknown about the button. The effect I have set up would keep the element hidden and new user would be able to come across full-screen button whenever he scrolls over the page.
I have also removed dialog box. Will be adding that only when setup of customization is done.

File: src/app/media-wall/media-wall-header/media-wall-header.component.html [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+				</a>
 			</div>
-		</a>
-		<div class="title">{{query}}</div>
+		<div class="title">
+			<span class="query">{{query}}
+			</span>
+			</div>
+		<div class="menu" [class.mouse-moving]="showHideMenu">
+			<div class="option" toggleFullscreen>
```
Comment (hemantjadon) [inline src/app/media-wall/media-wall-header/media-wall-header.component.html]: Wrap this inside a button element, as currently this button is not accessible
Always keep `a11y` in mind while writing components, also please make sure you provide proper role and name to the button.

**Comment by Achint08** [comment]
@hemantjadon done! Checkout now.

**Pull request merged**